### PR TITLE
Adds `min_layer_filters` and `padding_mode` parameters to mobilenet_v2

### PR DESCRIPTION
**Pull request by tomerk**
This commit adds a 'min_layer_filters' parameter to mobilenet_v2 as well as several padding modes to use for the mobilenet_v2 convolutions.

These are required to transition the Tensorflow Object Detection API (https://github.com/tensorflow/models/tree/master/research/object_detection) to using Keras.

CC'ing @fchollet @allenlavoie

### CODE REVIEW

File: keras_applications/mobilenet_v2.py [hunk 1]
```diff
@@ -161,6 +167,27 @@ def MobileNetV2(input_shape=None,
                  are used at each layer.
         depth_multiplier: depth multiplier for depthwise convolution
             (also called the resolution multiplier)
+        min_layer_filters: The minimum number of filters to use in any layer.
```
Comment (fchollet) [inline keras_applications/mobilenet_v2.py]: "min_filters" is a shorter name. Please make the description clearer, it isn't entirely clear how this argument works, from the description.

File: keras_applications/mobilenet_v2.py [hunk 2]
```diff
@@ -161,6 +167,27 @@ def MobileNetV2(input_shape=None,
...
+            This is applied after the width multiplier is used and the layer
+            filter count is made divisible by 8. If None (the default), no
+            additional minimum number of layers will be applied.
+        padding_mode: The padding mode to use for convolutions. One of
+          `mobilenet_v2.PADDING_ALL_SAME`,
+          `mobilenet_v2.PADDING_FRAMEWORK_COMPAT`,
+          `mobilenet_v2.PADDING_ALL_EXPLICIT`.
```
Comment (fchollet) [inline keras_applications/mobilenet_v2.py]: This is not a good API, I'm afraid. In a good API, the meaning of arguments should be understandable without having any context about the underlying implementation, only from domain-relevant information. Meaning that, in this case, someone who is familiar with MobileNet in general should immediately understand 1) what this argument does, 2) why this is an option in the constructor in the first place.

The API, the options made available to the user, should be about solving the user's problems, not about describing the underlying implementation, or routing around implementation issues.

In addition, in a good API, you don't need to use enums when strings work just as well, e.g. 'same', 'explicit'.

I believe this argument should not exist, as it does not provide value to a potential user.

**Comment by taehoonlee** [comment]
Thank you for the PR, @tomerk. I'll close the PR for now. Please feel free to open it again at any time if you have additional ideas for revisions.
